### PR TITLE
[StandardInstrumentation] Annotate loops with the function name

### PR DESCRIPTION
**Pull request by annamthomas**
When analyzing pass debug output it is helpful to have the function name along with the loop name.

### CODE REVIEW

**Comment by annamthomas** [comment]
I've not updated all the failing tests check lines, want to know this is okay before going forward.

**Comment by aeubanks** [comment]
makes sense to me, although bikeshedding a bit, I'd prefer
`Running pass: SimpleLoopUnswitchPass on loop %loop_begin in function test6`

happy to hear any other tweaks to what actually gets printed

**Comment by annamthomas** [comment]
> makes sense to me, although bikeshedding a bit, I'd prefer
Running pass: SimpleLoopUnswitchPass on loop %loop_begin in function test6

Yes, I can change to that. I'll wait a day to see if anyone has other preferences.

**Comment by jamieschmeiser** [comment]
I like the idea of making the loop easier to identify.  Have you considered demangling the function name also, or is that already handled in getName()?   However, that may make filtering difficult... what about printing both the mangled and demangled name if it is different?

**Comment by nikic** [comment]
> makes sense to me, although bikeshedding a bit, I'd prefer `Running pass: SimpleLoopUnswitchPass on loop %loop_begin in function test6`

+1

> I like the idea of making the loop easier to identify. Have you considered demangling the function name also, or is that already handled in getName()? However, that may make filtering difficult... what about printing both the mangled and demangled name if it is different?

I would not find demangling helpful here -- if I'm looking at debug logs, I want to line them up with IR, not with source.

**Comment by jamieschmeiser** [comment]
> I would not find demangling helpful here -- if I'm looking at debug logs, I want to line them up with IR, not with source.

I guess it depends on the situation as to whether demangling would be useful.  I find demangled names easier to understand but the filtering mechanisms work with mangled names.  This is why I suggested both but that may be too much info on the line.  At the risk of over-designing something simple, perhaps demangling could be controlled by a hidden option?  

Demangled or not, adding the function name is a good step.

**Comment by annamthomas** [comment]
Suggestion about formatting noted.
For now, I'm just going to be adding the function name to progress ahead with this (and general helps with the use cases I'm after).

File: llvm/test/Transforms/LoopUnroll/revisit.ll [hunk 1]
```diff
@@ -42,29 +42,29 @@ l0.0.1.ph:
...
   br i1 %cond.0.0.1, label %l0.0.1, label %l0.0.latch
-; CHECK: LoopFullUnrollPass on l0.0.1
+; CHECK: LoopFullUnrollPass on loop %l0.0.1
 ; CHECK-NOT: LoopFullUnrollPass
 
 l0.0.latch:
   %cmp = icmp slt i32 %iv.next, 2
   br i1 %cmp, label %l0.0, label %l0.latch
-; CHECK: LoopFullUnrollPass on l0.0
+; CHECK: LoopFullUnrollPass on loop %l0.0
```
Comment (aeubanks) [inline llvm/test/Transforms/LoopUnroll/revisit.ll:51]: the CHECK line isn't consistent with the ones you added below with the function name, I'd either keep everything without or with the function name in the CHECK

**Review by aeubanks — APPROVED** [review]
lgtm with comment addressed

File: llvm/test/Transforms/LoopUnroll/revisit.ll (hunk 1 above)
Comment (annamthomas) [reply llvm/test/Transforms/LoopUnroll/revisit.ll:51]: fixed. Also, the recommit tests showed some failures in lit tests. Fixed that as well.

**Comment by jamieschmeiser** [comment]
I had lost track of https://github.com/llvm/llvm-project/pull/87626 but the memory of it was haunting me during this review.  It hasn't landed yet but it may be something you want to look at.

**Pull request merged**